clientapi/routing/sendevent.go

```diff
@@ -120,6 +127,46 @@ func SendEvent(
 	return res
 }
 
+func checkPowerLevels(
+	ctx context.Context,
+	roomID string,
+	rsAPI api.RoomserverInternalAPI,
+	event *gomatrixserverlib.Event,
+) (ok bool, response *util.JSONResponse) {
+	req := api.QueryLatestEventsAndStateRequest{RoomID: roomID}
```

Comment:
We should probably be setting `StateToFetch` in `req` so that we aren't pulling out lots of state events that we don't need - ultimately we only care about the create event and the existing power level event.